### PR TITLE
Create expedited.cpp and move expedited code there

### PR DESCRIPTION
**Pull request by ptschip**

### CODE REVIEW

**Comment by sickpig** [comment]
@gandrewstone  I wonder if we could add `expedited.{cpp,h}` to the `src/.formatted-files`

**Comment by ptschip** [comment]
@sickpig thanks, i'll add those in to the list and push up.

File: src/expedited.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    EXPEDITED_TXNS = 4,
+};
+
+enum
+{
+    EXPEDITED_MSG_HDR = 1,
+    EXPEDITED_MSG_XTHIN = 2,
+};
+
+extern CCriticalSection cs_xpedited;
```
Comment (kyuupichan) [inline src/expedited.h]: Can you add a comment as to what precisely this critical section is intended to be protecting?

Comment (ptschip) [reply src/expedited.h]: ok done

File: src/expedited.cpp [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+#define NUM_XPEDITED_STORE 10
+
+// Just save the last few expedited sent blocks so we don't resend (uint256)
+uint256 xpeditedBlkSent[NUM_XPEDITED_STORE];
+
+// zeros on construction)
+int xpeditedBlkSendPos = 0;
+
+using namespace std;
+
```
Comment (kyuupichan) [inline src/expedited.cpp]: Is the using needed?  I see std:: throughout the file; let's be consistently explicit as that seems to be the trend.

File: src/expedited.cpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+#include <univalue.h>
+
+#define NUM_XPEDITED_STORE 10
+
+// Just save the last few expedited sent blocks so we don't resend (uint256)
+uint256 xpeditedBlkSent[NUM_XPEDITED_STORE];
+
+// zeros on construction)
+int xpeditedBlkSendPos = 0;
+
```
Comment (kyuupichan) [inline src/expedited.cpp]: I think these variables should be static; no point exporting them as they're file-local.

**Review by kyuupichan — CHANGES_REQUESTED** [review]
Apart from these minor points looks good to me.

**Comment by ptschip** [comment]
@kyuupichan fixed your 2 issues and squashed them into the previous commit...also cleaned up and replaced the boost::lexical casts.

**Comment by ptschip** [comment]
@kyuupichan in some cases also updated the use of error() messages instead of LogPrintf followed by return false.

**Comment by ptschip** [comment]
i really hate the way clang-format will indent the variables by 4 but not the text for print statements, so we end up with a strange looking indentations for error() or logprintf()

**Review by kyuupichan — APPROVED** [review]

**Comment by ptschip** [comment]
rebased

**Comment by ptschip** [comment]
@gandrewstone there is some kind of intermittent crash in the exploit_tests.cpp...it's hard to find out what it is because it only happens on Travis.  Pushed up another attempt at a fix.

**Comment by gandrewstone** [comment]
the exploit_tests.cpp travis issue isn't relevant to this PR AFAICT, so merging...

**Pull request merged**